state_machine:
  INITIALIZE:
    type: auv_smach.initialize.InitializeState
    parameters: {}

  NAVIGATE_THROUGH_GATE:
    type: auv_smach.gate.NavigateThroughGateState
    parameters:
      gate_depth: -1.5
      n_turns: -1

  NAVIGATE_AROUND_RED_BUOY:
    type: auv_smach.red_buoy.RotateAroundBuoyState
    parameters:
      radius: 2.2
      direction: 'ccw'
      red_buoy_depth: -0.7

  NAVIGATE_TO_TORPEDO_TASK:
    type: auv_smach.torpedo.TorpedoTaskState
    parameters:
      torpedo_map_radius: 1.5
      torpedo_map_depth: -1.3

  NAVIGATE_TO_BIN_TASK:
    type: auv_smach.bin.BinTaskState
    parameters:
      bin_whole_depth: -1.0

  NAVIGATE_TO_OCTAGON_TASK:
    type: auv_smach.octagon.OctagonTaskState
    parameters:
      octagon_depth: -1.0